File: app/src/main/java/org/flyontime/android/ui/animation/AnimationUtils.java
package org.flyontime.android.ui.animation;

import android.animation.Animator;
import android.animation.ValueAnimator;
import android.view.View;
import android.view.ViewGroup;

import org.flyontime.jw.android.R;

/**
 * Created by jossi on 18.06.2017.
 */

public class AnimationUtils {

    public static void expandView(View cardView, int height) {

        ValueAnimator anim = ValueAnimator.ofInt(cardView.getMeasuredHeightAndState(),
                height);
        anim.addUpdateListener(valueAnimator -> {
            int val = (Integer) valueAnimator.getAnimatedValue();
            ViewGroup.LayoutParams layoutParams = cardView.getLayoutParams();
            layoutParams.height = val;
            cardView.setLayoutParams(layoutParams);
        });
        anim.addListener(new Animator.AnimatorListener() {
            @Override
            public void onAnimationStart(Animator animation) {

            }

            @Override
            public void onAnimationEnd(Animator animation) {
                cardView.findViewById(R.id.revealContent).setVisibility(View.VISIBLE);
            }

            @Override
            public void onAnimationCancel(Animator animation) {

            }

            @Override
            public void onAnimationRepeat(Animator animation) {

            }
        });
        anim.start();

    }

    public static void collapseView(View cardView, int minHeight) {

        ValueAnimator anim = ValueAnimator.ofInt(cardView.getMeasuredHeightAndState(),
                minHeight);
        anim.addUpdateListener(valueAnimator -> {
            int val = (Integer) valueAnimator.getAnimatedValue();
            ViewGroup.LayoutParams layoutParams = cardView.getLayoutParams();
            layoutParams.height = val;
            cardView.setLayoutParams(layoutParams);

        });
        anim.addListener(new Animator.AnimatorListener() {
            @Override
            public void onAnimationStart(Animator animation) {

            }

            @Override
            public void onAnimationEnd(Animator animation) {
                cardView.findViewById(R.id.revealContent).setVisibility(View.GONE);
            }

            @Override
            public void onAnimationCancel(Animator animation) {

            }

            @Override
            public void onAnimationRepeat(Animator animation) {

            }
        });
        anim.start();
    }

}

File: app/src/main/java/org/flyontime/android/ui/animation/AnimationUtils.kt
package org.flyontime.android.ui.animation

import android.animation.Animator
import android.animation.ValueAnimator
import android.view.View
import org.flyontime.jw.android.R

/**
 * Created by jossi on 18.06.2017.
 */

object AnimationUtils {

    fun expandView(cardView: View, height: Int) {

        val anim = ValueAnimator.ofInt(cardView.measuredHeightAndState,
                height)
        anim.addUpdateListener { valueAnimator ->
            val `val` = valueAnimator.animatedValue as Int
            val layoutParams = cardView.layoutParams
            layoutParams.height = `val`
            cardView.layoutParams = layoutParams
        }
        anim.addListener(object : Animator.AnimatorListener {
            override fun onAnimationStart(animation: Animator) {

            }

            override fun onAnimationEnd(animation: Animator) {
                cardView.findViewById(R.id.revealContent).visibility = View.VISIBLE
            }

            override fun onAnimationCancel(animation: Animator) {

            }

            override fun onAnimationRepeat(animation: Animator) {

            }
        })
        anim.start()

    }

    fun collapseView(cardView: View, minHeight: Int) {

        val anim = ValueAnimator.ofInt(cardView.measuredHeightAndState,
                minHeight)
        anim.addUpdateListener { valueAnimator ->
            val `val` = valueAnimator.animatedValue as Int
            val layoutParams = cardView.layoutParams
            layoutParams.height = `val`
            cardView.layoutParams = layoutParams

        }
        anim.addListener(object : Animator.AnimatorListener {
            override fun onAnimationStart(animation: Animator) {

            }

            override fun onAnimationEnd(animation: Animator) {
                cardView.findViewById(R.id.revealContent).visibility = View.GONE
            }

            override fun onAnimationCancel(animation: Animator) {

            }

            override fun onAnimationRepeat(animation: Animator) {

            }
        })
        anim.start()
    }

}
